### DISCUSSION TITLE
PHPCR : Missing root node properties

### DISCUSSION THREAD

**Discussion by Adelysnet** [sulu/sulu#6596] [Q&A]
Hi,

I don't know how.. wrong build.. change in webspace config but it ends up all my website's routes suddenly stopped working, showing a "No route found exeption" on the front. 

After deep searches, I notice that all the sulu properties (sulu:content|created|changed|history) where missing on the live session of my repository (with jackrabbit). I had to create them manually in phpcr shell (changing route from backend also works, except for the homepage), exporting them from the default session and importing again to the live one. 

My question is, beside I can't tell if this is a bug or a bad manipulation of my own, is there any simpler way to rebuild the routes properties ?

Sulu Version : 2.4.2
Jackrabbit version : 2.20.5
Symfony Version : 5.4.8
PHP Version : 7.4.26

**Comment by alexander-schranz** [comment]
If you change the webspace key you need to migrate it also in your phpcr. If you do not yet have any data in your database. You can just go with `bin/console sulu:build dev --destroy` to have a clean installation of it. This really purge the whole database so carefully with it.

If you just want to create the new webspace nodes you need to run `bin/console sulu:document:init` like documented in the adding new webpsace documentation: https://docs.sulu.io/en/2.4/cookbook/create-new-webspace.html

**Reply by Adelysnet** [reply]
I did not change the webspace key, my content was still there, only the sulu routes properties were missing. By the way, I tried to remove the property `sulu:content` from my root node '`/cmf/{project}/routes/{locale}`, save and launch the given command `sulu:document:init`, it did not recreate the missing property `sulu:content`. Should i have deleted the route instead ? 

 I think I came through this after adding portals in my webspace, but not sure, if the problem arises, I'll report the steps to reproduce it.

Thanks for your help.

**Reply by Adelysnet** [reply]
Update : when i execute the commande `bin/console sulu:document:init`, `route` and `homepage` have empty brackets  in console output : 

`sulu_core.webspace.document_manager.webspace_initializer
[ ] homepage: /cmf/{project}/contents (fr)
[ ] route: /cmf/{project}/routes/{locale} (fr)`

Does it mean nothing was done for this nodes ?
